api/groups_test.go

```diff
@@ -257,3 +258,243 @@ func TestAddUserToGroupHandler(t *testing.T) {
 		}
 	})
 }
+
+func TestRemoveUserFromGroupHandler(t *testing.T) {
+	var (
+		ctx                                                                                = context.Background()
+		userNotFoundDescription, groupNotFoundDescription, internalErrorDescription string = "user not found", "group not found", "internal error"
+	)
+
+	api, w, m := apiSetup()
+
+	timeStamp := time.Now()
+	getGroupData := &cognitoidentityprovider.GroupType{
+		Description:  aws.String("a test group"),
+		GroupName:    aws.String("test-group"),
+		Precedence:   aws.Int64(100),
+		CreationDate: &timeStamp,
+	}
+
+	Convey("Remove a user from a group - check expected responses", t, func() {
+		adminCreateUsersTests := []struct {
+			removeUserFromGroupFunction func(userInput *cognitoidentityprovider.AdminRemoveUserFromGroupInput) (*cognitoidentityprovider.AdminRemoveUserFromGroupOutput, error)
+			getGroupFunction            func(input *cognitoidentityprovider.GetGroupInput) (*cognitoidentityprovider.GetGroupOutput, error)
+			listUsersForGroupFunction   func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error)
+			assertions                  func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse)
+		}{
+			// 202 response - user removed to group
+			{
+				func(userInput *cognitoidentityprovider.AdminRemoveUserFromGroupInput) (*cognitoidentityprovider.AdminRemoveUserFromGroupOutput, error) {
+					return &cognitoidentityprovider.AdminRemoveUserFromGroupOutput{}, nil
+				},
+				func(inputData *cognitoidentityprovider.GetGroupInput) (*cognitoidentityprovider.GetGroupOutput, error) {
+					group := &cognitoidentityprovider.GetGroupOutput{
+						Group: getGroupData,
+					}
+					return group, nil
+				},
+				func(input *cognitoidentityprovider.ListUsersInGroupInput) (*cognitoidentityprovider.ListUsersInGroupOutput, error) {
+					return &cognitoidentityprovider.ListUsersInGroupOutput{}, nil
+				},
+				func(successResponse *models.SuccessResponse, errorResponse *models.ErrorResponse) {
+					So(successResponse, ShouldNotBeNil)
+					So(successResponse.Status, ShouldEqual, http.StatusOK)
+					So(errorResponse, ShouldBeNil)
+				},
+			},
+			// Cognito 404 response - user not found
```

Comment:
Looking at the response, and the swagger doc, I think this should be `400`

Reply:
This comment is describing the response from Cognito not the response from the API.
Would probably be less confusing to leave the Cognito status code out of the comment

Reply:
I think I've misunderstood your comment here.

Reply:
My bad - its the 404 from Cognito